src/test/java/io/jenkins/pluginhealth/scoring/service/UpdateCenterServiceTest.java

```diff
@@ -30,22 +30,28 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
+import io.jenkins.pluginhealth.scoring.AbstractDBContainerTest;
 import io.jenkins.pluginhealth.scoring.model.Plugin;
 
 import org.junit.jupiter.api.Test;
-import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
 import org.springframework.boot.test.autoconfigure.json.JsonTest;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Service;
 
-@JsonTest
-@RunWith(SpringRunner.class)
-class UpdateCenterServiceTest {
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@DataJpaTest(includeFilters = @ComponentScan.Filter(Service.class))
+class UpdateCenterServiceTest extends AbstractDBContainerTest {
```

Comment:
As per my understanding, the tests were failing because now I'm using @Autowired annotation in this test so I need to build the application by extending this class to `AbstractDBContainerTest` and use `@DataJpaTest` to not load the entire application.

I removed the @JsonTest and @RunWith annotation because the tests pass without them, but I'm not sure this is the right decision so I wanted to confirm once.

Reply:
with https://github.com/jenkins-infra/plugin-health-scoring/pull/20#discussion_r930954167, we wouldn't have to make a DB connection to test the `UpdateCenterService`.